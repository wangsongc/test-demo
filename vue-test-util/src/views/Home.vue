<template>
  <div class="wrapper">
    <app-header></app-header>
    <app-sidebar></app-sidebar>
    <div class="content" :class="{'content--collapse':collapse}">
      <router-view></router-view>
    </div>
  </div>
</template>
<script>
import AppHeader from '@/components/Header.vue'
import AppSidebar from '@/components/Sidebar.vue'
import eventHub from '@/assets/js/EventHub.js'
import { checkWidth } from '@/assets/js/CommonMethods.js'
export default {
  name: 'layout',
  components: {
    AppHeader,
    AppSidebar
  },
  data () {
    return {
      collapse: checkWidth()
    }
  },
  methods: {},
  created () {
    eventHub.$on('collapse', msg => {
      this.collapse = msg
    })
  },
  mounted () {},
  watch: {},
  computed: {}
}
</script>
<style>
.content {
  position: absolute;
  left: 230px;
  right: 0;
  top: 70px;
  bottom: 0;
  width: auto;
  padding: 32px;
  overflow-y: auto;
  overflow-x: hidden;
  -webkit-transition: left 0.3s ease-in-out;
  transition: left 0.3s ease-in-out;
}
.content--collapse {
  left: 65px;
}

@media screen and (max-width: 1025px) {
  .content {
    left: 65px;
  }
}

@media screen and (max-width: 769px) {
  .content {
    padding: 12px 12px 30px;
  }
}
</style>
