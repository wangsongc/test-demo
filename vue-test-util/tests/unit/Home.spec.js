/**
 * Home.vue组件
 * 测试内容包括以下
 * 挂载子组件-stubs
 * 临时vue实例上挂载vue-router
 */
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Home from '@/views/Home.vue'
import VueRouter from 'vue-router'
import eventHub from '@/assets/js/EventHub.js'

// 创建临时Vue实例，挂载组件中使用的插件
const localVue = createLocalVue()
localVue.use(VueRouter)

describe('Home.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(Home, { localVue,
      stubs: ['app-header', 'app-sidebar']
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  // 测试内容：eventHub $on
  it('eventHub $on test', () => {
    wrapper.vm.collapse = false
    // 顺手测试一下DOM结构中类名的存在与否
    expect(wrapper.find('.content').classes()).not.toContain('content--collapse')
    eventHub.$emit('collapse', true)
    expect(wrapper.vm.collapse).toBeTruthy()
    // 顺手测试一下DOM结构中类名的存在与否
    expect(wrapper.find('.content').classes()).toContain('content--collapse')
  })

  // 测试内容：snapshot->概括的测试DOM结构
  // 如果组件内存在比较特殊的需要测试的DOM结构的话，可以单独测试（详见AppButton测试文件末尾）
  it('snapshot test', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
