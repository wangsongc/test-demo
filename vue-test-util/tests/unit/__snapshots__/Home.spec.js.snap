// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue snapshot test 1`] = `
<div class="wrapper">
  <app-header-stub></app-header-stub>
  <app-sidebar-stub></app-sidebar-stub>
  <div class="content content--collapse">
    <router-view-stub name="default"></router-view-stub>
  </div>
</div>
`;
